pallets/msa/src/lib.rs

```diff
@@ -1090,8 +1091,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Removes all delegations from the specified delegator MSA id to providers
-	pub fn delete_delegation_relationship(delegator: DelegatorId) {
-		_ = DelegatorAndProviderToDelegation::<T>::clear_prefix(delegator, u32::max_value(), None);
+	pub fn delete_delegation_relationship(delegator: DelegatorId) -> u32 {
+		let result = DelegatorAndProviderToDelegation::<T>::clear_prefix(
+			delegator,
+			MAX_NUMBER_OF_PROVIDERS_PER_DELEGATOR,
+			None,
+		);
+		result.backend
```

Comment:
I'm not sure this should be backend. I think it should be unique but I'm not certain

Reply:
I've modified to return the number of unique so let's re-evaluate it.